test/DumberComponent_test.js

```diff
@@ -0,0 +1,33 @@
+const React = require('react');
+const { shallow } = require('enzyme');
+const sinon = require('sinon');
+
+const Noop = (props) => { return <p>Noop</p> };
+const DumberComponent = require('../components/DumberComponent');
+
+describe('<DumberComponent />', () => {
+  let wrapper, spy;
+
+  beforeEach(() => {
+    spy = sinon.spy();
+    wrapper = !DumberComponent.prototype ?
+      shallow(<Noop />) : shallow(<DumberComponent handleClick={spy} />);
+  });
+
+  it('should not be a class-based component', () => {
+    const tryToGetState = () => { wrapper.state(); }
+    expect(tryToGetState).toThrow(
+      'ShallowWrapper::state() can only be called on class components'
```

Comment:
I'd 'humanize' this error output a little more: maybe a simple '<DumberComponent />' should be a regular function, or something similar?